oteapi/datacache/datacache.py

```diff
@@ -77,7 +77,15 @@ async def async_func(args):
         result = await future
         return result
 
-    return asyncio.run(async_func(args))
+    try:
+        # Will raise a RuntimeError if there is no event loop
+        asyncio.get_running_loop()
+    except RuntimeError:
+        # no event loop
+        return asyncio.run(async_func(args))
+    else:
+        # within a running event loop
+        return func(*args)
```

Comment:
If a loop is already running, couldn't we simply run the function in `loop.run_in_executor(None, func, *args)`? Making it run in the current loop?